WordPress/gutenberg
# Font Library: Refactor Font Family REST API controller with new design

## What?

Refactors `WP_REST_Font_Families_Controller` to implement [the proposed design](https://github.com/WordPress/gutenberg/issues/57616).

## Why?

This is another step toward refactoring the Font Library in preparation for merging to Core. See https://github.com/WordPress/gutenberg/issues/55278


## Testing Instructions

Test the following following endpoints, which are added or revised from the previous controller design, to ensure the work as expected in the design and have appropriate error handling.

```
GET wp/v2/font-families
GET wp/v2/font-families/<id>
GET wp/v2/font-families/<id>?_embed=true
POST wp/v2/font-families
POST wp/v2/font-families/<id>
DELETE wp/v2/font-families/<id>
DELETE /font-families/1?force=true
```

One thing this endpoint does not currently handle is deleting child font faces when the family is deleted. I plan to handle that in a follow-up PR.
